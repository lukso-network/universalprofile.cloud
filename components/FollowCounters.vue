<script setup lang="ts">
const { formatMessage, formatNumber } = useIntl()
const { showModal } = useModal()

type Props = {
  profileFollowers?: ProfileFollowers
}

const props = defineProps<Props>()

const handleShowFollowing = () => {
  showModal({
    template: 'Followers',
    data: {
      addresses: props.profileFollowers?.followingAddresses,
      count: props.profileFollowers?.followingCount,
      type: 'following',
    },
    size: 'medium',
  })
}

const handleShowFollowers = () => {
  showModal({
    template: 'Followers',
    data: {
      addresses: props.profileFollowers?.followerAddresses,
      count: props.profileFollowers?.followerCount,
      type: 'follower',
    },
    size: 'medium',
  })
}
</script>

<template>
  <AppPlaceholderLine
    v-if="profileFollowers?.isLoading"
    class="h-[20px] w-[160px]"
  />
  <div
    v-else
    class="paragraph-inter-12-medium group flex cursor-pointer items-center"
  >
    <div
      class="rounded-l-4 border border-neutral-90 px-1.5 transition hover:border-neutral-20 group-hover:border-r-neutral-20"
      @click="handleShowFollowing"
    >
      <span class="paragraph-inter-12-bold">{{
        formatNumber(profileFollowers?.followingCount || 0)
      }}</span>
      {{ formatMessage('profile_card_following') }}
    </div>
    <div
      class="rounded-r-4 border border-l-0 border-l-neutral-90 px-1.5 transition hover:border-neutral-20"
      @click="handleShowFollowers"
    >
      <span class="paragraph-inter-12-bold">{{
        formatNumber(profileFollowers?.followerCount || 0)
      }}</span>
      {{ formatMessage('profile_card_followers') }}
    </div>
  </div>
</template>
