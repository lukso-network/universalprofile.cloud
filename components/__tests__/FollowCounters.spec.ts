import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import FollowCounters from '../FollowCounters.vue'

import type { ProfileFollowers } from '@/types/profile'

describe('FollowCounters', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render empty state', async () => {
    const component = await renderSuspended(FollowCounters)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render counters', async () => {
    const component = await renderSuspended(FollowCounters, {
      props: {
        profileFollowers: {
          followerCount: 1000,
          followingCount: 2000,
        } as ProfileFollowers,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
