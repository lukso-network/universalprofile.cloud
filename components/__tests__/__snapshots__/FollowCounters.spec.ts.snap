// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FollowCounters > should render counters 1`] = `
"<div id="test-wrapper">
  <div class="paragraph-inter-12-medium group flex cursor-pointer items-center">
    <div class="rounded-l-4 border border-neutral-90 px-1.5 transition hover:border-neutral-20 group-hover:border-r-neutral-20"><span class="paragraph-inter-12-bold">2,000</span> Following</div>
    <div class="rounded-r-4 border border-l-0 border-l-neutral-90 px-1.5 transition hover:border-neutral-20"><span class="paragraph-inter-12-bold">1,000</span> Followers</div>
  </div>
</div>"
`;

exports[`FollowCounters > should render empty state 1`] = `
"<div id="test-wrapper">
  <div class="paragraph-inter-12-medium group flex cursor-pointer items-center">
    <div class="rounded-l-4 border border-neutral-90 px-1.5 transition hover:border-neutral-20 group-hover:border-r-neutral-20"><span class="paragraph-inter-12-bold">0</span> Following</div>
    <div class="rounded-r-4 border border-l-0 border-l-neutral-90 px-1.5 transition hover:border-neutral-20"><span class="paragraph-inter-12-bold">0</span> Followers</div>
  </div>
</div>"
`;
